lib/translator/common.cc

```diff
@@ -54,6 +54,21 @@ void SetAllocPageCallbacks(uint64_t (*alloc_callback)(uint16_t),
   dealloc_pages_callback = dealloc_callback;
 }
 
+StatusCode Allocation::SetPages(uint16_t data_page_count,
+                                uint16_t mdata_page_count) {
+  this->data_page_count = data_page_count;
+  this->data_addr = AllocPages(data_page_count);
+  this->mdata_page_count = mdata_page_count;
+  this->mdata_addr = AllocPages(mdata_page_count);
+
+  if (this->data_addr == 0 || this->mdata_addr == 0) {
```

Comment:
What if mdata_page_count is set to 0 and the desired behavior is for the mdata_addr to be 0? We probably need a more "conditional" approach to this

Reply:
that's a good point. Should I also account for the case where data_page_count is set to 0? Or is that not a meaningful situation?

Reply:
I think it could potentially be a meaningful situation; however, I can't think of any cases off the top of my head that would require that functionality. Nonetheless, it's probably good practice to handle both cases synonymously